### PR TITLE
Disable forced test selector stripping in addon templates

### LINKED ISSUE #95: Discussion: ability to not strip test-selectors in addon builds?
Here's our situation: we are extracting an addon out of app. The tests for this app use ember-test-selectors. Once extracted to an addon, though, the same tests do not pass because when the addon builds, test selectors are stripped.

It would seem to be a inarguably good thing to allow addon authors to use ember-test-selectors to test their addons.

In addition, I would like to propose that apps be able to not strip test selectors when building their addon for tests, either by default or by configuration.

Thoughts? Is this already possible and I am missing it?

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves #95

/cc @lukemelia @rwjblue

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -26,32 +26,36 @@ module.exports = {
...
   },
 
   included(app) {
     this._super.included.apply(this, arguments);
+    this._ensureFindHost();
+
+    let host = this._findHost();
+    this._assignOptions(host);
+
+    // we can't use the setupPreprocessorRegistry() hook as it is called to
```
Comment (rwjblue) [inline index.js]: FWIW, this will be fixed in https://github.com/ember-cli/ember-cli/pull/7059.

Comment (Turbo87) [inline index.js]: ohhh... so that's what that PR is for 😄

File: index.js [hunk 2]
```diff
@@ -26,32 +26,36 @@ module.exports = {
...
-          plugin: StripTestSelectorsTransform,
-          baseDir() { return __dirname; }
-        });
-      } else {
-        let TransformTestSelectorParamsToHashPairs = require('./transform-test-selector-params-to-hash-pairs');
+  _setupPreprocessorRegistry(registry) {
+    if (this._stripTestSelectors) {
+      let StripTestSelectorsTransform = require('./strip-test-selectors');
+
+      registry.add('htmlbars-ast-plugin', {
```
Comment (rwjblue) [inline index.js]: Seems like this could be made a bit simpler/easy to read:

```js
let transform = 'transform-test-selector-params-to-hash-pairs';

if (this._stripTestSelectors) {
  transform = 'strip-test-selectors';
}

registry.add('htmlbars-ast-plugin', {
  name: transform,
  plugin: require(`./${transform}`),
  baseDir() { return __dirname; }
});
```

Comment (Turbo87) [inline index.js]: sure, seems like an unrelated refactoring though :)

**Comment by lukemelia** [comment]
Thanks @Turbo87!!! 💯
